filters/HAGFilter.cpp

```diff
@@ -136,103 +213,88 @@ void HAGFilter::filter(PointView& view)
             "as ground");
 
     // Build the 2D KD-tree.
-    KD2Index& kdi = gView->build2dIndex();
+    const KD2Index& kdi = gView->build2dIndex();
 
     // Second pass: Find Z difference between non-ground points and the nearest
     // neighbor (2D) in the ground view or between non-ground points and the
     // locally-computed surface (Delaunay triangultion of the neighborhood).
     for (PointId i = 0; i < ngView->size(); ++i)
     {
         PointRef point = ngView->point(i);
-        double x0 = point.getFieldAs<double>(Dimension::Id::X);
-        double y0 = point.getFieldAs<double>(Dimension::Id::Y);
-        double z0 = point.getFieldAs<double>(Dimension::Id::Z);
-        auto ids = kdi.neighbors(point, m_count);
-        double z1 = std::numeric_limits<double>::quiet_NaN();
+        double x0 = point.getFieldAs<double>(Id::X);
+        double y0 = point.getFieldAs<double>(Id::Y);
+        double z0 = point.getFieldAs<double>(Id::Z);
+        PointIdList ids = kdi.neighbors(point, m_count);
```

Comment:
If we use `knnSearch`, then we get the squared distances for free and do not need to recalculate within the loop below.